### PR TITLE
Document relationship between upsert sinks and primary keys

### LINKED ISSUE #745: Document the relationship between the primary key and upsert sink
The original Slack thread is in Chinese, let me translate it:
https://risingwave-labs.slack.com/archives/C04F4RB1AHF/p1678950795647739

1. If the downstream system supports the primary key, then RW does not allow users to define the primary key in the sink.
2. If the downstream system does not support the primary key, then RW forces users to define the primary key in the sink.
3. If the table in the downstream system has no primary key, then RW does not allow users to define an `upsert` sink.

@xx01cyx Correct me if I am wrong thanks!

~I think these are a set of general rules that apply to all kinds of sinks/downstream systems,
therefore it seems that it can be mentioned in some `overview` page
and have a table that lists whether each type of downstream system supports the primary key or not~

### LINKED ISSUE #937: Document Sinks Overview lacks an introduction about Apache Iceberg
### Where's the bug?

https://www.risingwave.dev/docs/current/delivery-overview

### Describe the bug

The [overview page](https://www.risingwave.dev/docs/current/delivery-overview) might need some introduction about Apache Iceberg https://www.risingwave.dev/docs/current/sink-to-iceberg/.

### PR DESCRIPTION
**Pull request by emile-00**
## Info
- **Description**: 
Document relationship between upsert sinks and primary keys in data delivery overview page. Direct readers to overview page if sink guides if creating upsert sink.

- **Preview**: 
[ Paste the preview link to the edited page here. Edit this item after you submit the PR and the preview is ready. ]

- **Related code PR**: 
[ Provide the link to the code PR here. Delete this part if there's no code PR related. ]

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/745

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

**Comment by emile-00** [comment]
@xx01cyx Can you ensure that the updates made to Delivery Overview regarding upsert sinks and primary keys are accurate? Thank you!

**Review by xx01cyx — APPROVED** [review]
LGTM!

File: docs/guides/create-sink-kafka.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ All WITH options are required except `force_append_only` and `primary_key`.
 |connector| Sink connector type. Currently, only `‘kafka’` and `‘jdbc’` are supported. If there is a particular sink you are interested in, see [Integrations](/rw-integration-summary.md) for a full list of connectors and integrations we are working on. |
 |properties.bootstrap.server|Address of the Kafka broker. Format: `‘ip:port’`. If there are multiple brokers, separate them with commas. |
 |topic|Address of the Kafka topic. One sink can only correspond to one topic.|
-|type|Data format. Allowed formats:<ul><li> `append-only`: Output data with insert operations.</li><li> `debezium`: Output change data capture (CDC) log in Debezium format.</li><li> `upsert`: Output data as a changelog stream. `primary_key` must be specified in this case. </li></ul>|
+|type|Data format. Allowed formats:<ul><li> `append-only`: Output data with insert operations.</li><li> `debezium`: Output change data capture (CDC) log in Debezium format.</li><li> `upsert`: Output data as a changelog stream. `primary_key` must be specified in this case. </li></ul> If creating an `upsert` sink, see the [Overview](../delivery-overview.md) on when to define the primary key.|
```
Comment (hengm3467) [inline docs/guides/create-sink-kafka.md]: --> To learn about how to deine the primary key when creating an `upsert` sink, see []().

**Review by hengm3467 — APPROVED** [review]
Otherwise looks good to me. After the comment is implement, you can merge the PR. Thanks.

**Pull request merged**